SEGV running "zc -C" on Zed containing "from file"

Repro is with Zed commit 77bdc81.

```
$ zq -version
Version: v0.29.0-439-g77bdc815

$ zc -C 'from ( file m1.zson; )'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x13d5631]

goroutine 1 [running]:
github.com/brimdata/zed/zfmt.(*canon).proc(0xc00019fb20, 0x159f520, 0x0)
	/Users/phil/work/zed/zfmt/ast.go:183 +0xb1
github.com/brimdata/zed/zfmt.(*canon).proc(0xc00019fb20, 0x159f280, 0xc00015b8f0)
	/Users/phil/work/zed/zfmt/ast.go:244 +0xb1f
github.com/brimdata/zed/zfmt.(*canon).proc(0xc00019fb20, 0x159f520, 0xc00015b950)
	/Users/phil/work/zed/zfmt/ast.go:184 +0x108
github.com/brimdata/zed/zfmt.AST(0x159f520, 0xc00015b950, 0x1364d65, 0x143cb80)
	/Users/phil/work/zed/zfmt/ast.go:13 +0x7e
github.com/brimdata/zed/cmd/zed/compile.procFmt(0x159f520, 0xc00015b950, 0x13b7a01, 0x1435700, 0xc00015b110, 0x159f520, 0xc00015b950)
	/Users/phil/work/zed/cmd/zed/compile/command.go:302 +0xe9
github.com/brimdata/zed/cmd/zed/compile.(*Command).writeProc(0xc00008ab40, 0x159f520, 0xc00015b950)
	/Users/phil/work/zed/cmd/zed/compile/command.go:215 +0x49
github.com/brimdata/zed/cmd/zed/compile.(*Command).parse(0xc00008ab40, 0x7ffeefbff9e6, 0x16, 0x7ffeefbff9e6, 0x16)
	/Users/phil/work/zed/cmd/zed/compile/command.go:176 +0x465
github.com/brimdata/zed/cmd/zed/compile.(*Command).Run(0xc00008ab40, 0xc0002a6d40, 0x1, 0x1, 0x0, 0x0)
	/Users/phil/work/zed/cmd/zed/compile/command.go:141 +0x2ad
github.com/brimdata/zed/pkg/charm.path.run(0xc0002b4130, 0x2, 0x2, 0xc0002a6d40, 0x1, 0x1, 0xc0002b4130, 0x2)
	/Users/phil/work/zed/pkg/charm/path.go:11 +0x8f
github.com/brimdata/zed/pkg/charm.(*Spec).ExecRoot(0x17c8de0, 0xc0002a6d20, 0x3, 0x3, 0x2, 0x2)
	/Users/phil/work/zed/pkg/charm/charm.go:63 +0x1e7
main.main()
	/Users/phil/work/zed/cmd/zc/main.go:14 +0x193
```

--- a/zfmt/ast.go
+++ b/zfmt/ast.go
@@ -234,22 +234,22 @@ func (c *canon) proc(p ast.Proc) {
 		//XXX cleanup for len(Trunks) = 1
 		c.next()
 		c.open("from (")
-		c.open()
 		for _, trunk := range p.Trunks {
 			c.ret()
 			c.source(trunk.Source)
-			c.write(" => ")
-			c.open()
-			c.head = true
-			c.proc(trunk.Seq)
+			if trunk.Seq != nil {
+				c.write(" =>")
+				c.open()
+				c.head = true
+				c.proc(trunk.Seq)
+				c.close()
+			}
 			c.write(";")
-			c.close()
 		}
 		c.close()
 		c.ret()
 		c.flush()
 		c.write(")")
-		c.close()
 	case *ast.Const:
 		c.write("const %s=", p.Name)
 		c.expr(p.Expr, false)

--- a/zfmt/ztests/from.yaml
+++ b/zfmt/ztests/from.yaml
@@ -1,0 +1,40 @@
+script: |
+  zc -C 'file path'
+  echo ===
+  zc -C 'get http://host/path'
+  echo ===
+  zc -C 'from pool'
+  echo ===
+  zc -C 'from ( file path; get http://host/path; pool; )'
+  echo ===
+  zc -C 'from ( file path => head; get http://host/path => head; pool => head; )'
+
+outputs:
+  - name: stdout
+    data: |
+      from (
+        file path;
+      )
+      ===
+      from (
+        get http://host/path;
+      )
+      ===
+      from (
+        pool;
+      )
+      ===
+      from (
+        file path;
+        get http://host/path;
+        pool;
+      )
+      ===
+      from (
+        file path =>
+          head 1;
+        get http://host/path =>
+          head 1;
+        pool =>
+          head 1;
+      )

